python/dataproc_templates/gcs/gcs_to_spanner.py

```diff
@@ -0,0 +1,160 @@
+      help='Spanner output instance'
+    )
+    parser.add_argument(
+      f'--{constants.GCS_SPANNER_OUTPUT_DATABASE}',
+      dest=constants.GCS_SPANNER_OUTPUT_DATABASE,
+      required=True,
+      help='Spanner output database'
+    )
+    parser.add_argument(
+      f'--{constants.GCS_SPANNER_OUTPUT_TABLE}',
+      dest=constants.GCS_SPANNER_OUTPUT_TABLE,
+      required=True,
+      help='Spanner output table'
+    )
+    parser.add_argument(
+      f'--{constants.GCS_SPANNER_PRIMARY_KEY}',
+      dest=constants.GCS_SPANNER_PRIMARY_KEY,
+      required=False,
+      help=(
+        "Primary key of the table. "
+        "It is only required when the table doesn't exist"
+      )
+    )
+    parser.add_argument(
+      f'--{constants.GCS_SPANNER_OUTPUT_SAVE_MODE}',
+      dest=constants.GCS_SPANNER_OUTPUT_SAVE_MODE,
+      required=False,
+      default=constants.OUTPUT_MODE_APPEND,
+      help=(
+        'Output write mode '
+        '(one of: <Overwrite|ErrorIfExists|Append|Ignore>) '
+        '(Defaults to ErrorIfExists)'
+      ),
+      choices=[
+        constants.OUTPUT_MODE_OVERWRITE,
+        constants.OUTPUT_MODE_APPEND,
+        constants.OUTPUT_MODE_IGNORE,
+        constants.OUTPUT_MODE_ERRORIFEXISTS
+      ]
+    )
+    parser.add_argument(
+      f'--{constants.GCS_SPANNER_OUTPUT_BATCH_INSERT_SIZE}',
+      dest=constants.GCS_SPANNER_OUTPUT_BATCH_INSERT_SIZE,
+      required=False,
+      default=1000,
+      help='Spanner batch insert size '
+    )
+
+    known_args: argparse.Namespace
+    known_args, _ = parser.parse_known_args(args)
+
+    return vars(known_args)
+
+  def run(self, spark: SparkSession, args: Dict[str, Any]) -> None:
+
+    project_id: str = constants.PROJECT_ID_PROP
+    logger: Logger = self.get_logger(spark=spark)
+
+    # Arguments
+    input_file_location: str = args[constants.GCS_SPANNER_INPUT_LOCATION]
+    input_file_format: str = args[constants.GCS_SPANNER_INPUT_FORMAT]
+    spanner_instance: str = args[constants.GCS_SPANNER_OUTPUT_INSTANCE]
+    spanner_database: str = args[constants.GCS_SPANNER_OUTPUT_DATABASE]
+    spanner_table: str = args[constants.GCS_SPANNER_OUTPUT_TABLE]
+    spanner_pk: str = args[constants.GCS_SPANNER_PRIMARY_KEY]
+    spanner_save_mode: str = args[constants.GCS_SPANNER_OUTPUT_SAVE_MODE]
+    batch_size: str = args[constants.GCS_SPANNER_OUTPUT_BATCH_INSERT_SIZE]
+    spanner_url: str = (
+      f"jdbc:cloudspanner:/projects/{project_id}/"
+      f"instances/{spanner_instance}/"
+      f"databases/{spanner_database}"
+    )
+    spanner_driver = "com.google.cloud.spanner.jdbc.JdbcDriver"
+    table_options = f"PRIMARY KEY ({spanner_pk})" if spanner_pk else None
+
+    logger.info(
+      "Starting GCS to Spanner spark job with parameters:\n"
+      f"{pprint.pformat(args)}"
+    )
+
+    # Read
+    input_data: DataFrame
+
+    input_data = (
+      spark.read
+      .format(input_file_format)
+      .load(input_file_location)
+    )
+
+    # Write
+    (
```

Comment:
Can we change the formatting to be consistent with the rest of the templates?

An example from GCS to BigQuery:

https://github.com/GoogleCloudPlatform/dataproc-templates/blob/f864cd97ca96f5e3da99931f323d7c9a52035094/python/dataproc_templates/gcs/gcs_to_bigquery.py#L134-L139

Reply:
Yeah, we need a linter urgently. In the meantime I will change it for consistency but I strongly prefer the way I did it.

Reply:
Pylint is configured for the project, should just be a matter of setting it up in your IDE

Reply:
Done in cc4d05f